async/tls_async.mli

```diff
@@ -9,6 +9,23 @@ module Session = Session
     certificates, such as loading from a Unix filesystem *)
 module X509_async = X509_async
 
+(** [upgrade_server_handler] is what [listen] calls to handle each client.
+    It is exposed so that low-level end-users of the library can use tls-async
+    inside of code that manages Tcp services directly.
```

Comment:
that documentation string is now attached to the wrong value, and since it refers to listen, maybe push it after the listen value?

Reply:
ah, good catch. moved it below listen as well.